### DISCUSSION TITLE
mini.files - customize height of windows

### DISCUSSION THREAD

**Discussion by adriankarlen** [nvim-mini/mini.nvim#1140] [Q&A] [answered]
Hi, I am trying to modify the height of the windows in mini.files to all be the same height.

It feels like this should be doable using `MiniFilesWindowUpdate`-event. But I am unsure of how mini.files updates the height of the window. I have currently added the following snippet.

```lua
vim.api.nvim_create_autocmd("User", {
  pattern = "MiniFilesWindowUpdate",
  callback = function(args)
    local win_id = args.data.win_id

    local config = vim.api.nvim_win_get_config(win_id)
    config.height = 10
    vim.api.nvim_win_set_config(win_id, config)
  end,
})
```

This is the behavior currently.

https://github.com/user-attachments/assets/8fe3981d-8eb9-4044-ae1b-1b9ac8487577

As can be seen the event fires when opening the explorer, and when opening a file. Perhaps I am misunderstanding the way that the event is working? It is working as expected when settign stuff like border or winblend, is it that the plugin is setting the window height after the event has happened?

**Comment by echasnovski** [comment]
Thanks for reporting this!

Yeah, I think I'd also expect setting window height to work here.

The quick test showed that the culprit is an [intentional blocking of `MiniPickWindowUpdate`](https://github.com/echasnovski/mini.nvim/blob/7c00ea1a6b35332746e1a2fa768e26c5fa9c3589/lua/mini/files.lua#L1846-L1848) event to reduce it getting triggered during cursor movement. Maybe having this conditioned on whether there is a preview is a better approach.

There is also a [special treatment of window height](https://github.com/echasnovski/mini.nvim/blob/7c00ea1a6b35332746e1a2fa768e26c5fa9c3589/lua/mini/files.lua#L1862-L1871) that needs to be addressed for this autocommand to work. I'll take a look.

**Comment by echasnovski** [comment, marked as answer]
@adriankarlen, this now should be possible on latest `main` with [this documented example](https://github.com/echasnovski/mini.nvim/blob/974ebc04dff3b59899a5eee29157a9bbfd85dce3/doc/mini-files.txt#L393-L411). It not working previously was indeed the result of an intentional blocking of `MiniPickWindowUpdate` event from being triggered too frequently. As this is not the first issue that came out of this, it seems that triggering the event whenever window config changes in the better way to actually make it work as it was designed (to customize window config). There even is now a [test for this particular setup](https://github.com/echasnovski/mini.nvim/blob/974ebc04dff3b59899a5eee29157a9bbfd85dce3/tests/test_files.lua#L3870-L3877), so there is less chance that it stops working.

**Reply by adriankarlen** [reply]
Thank you for the very quick fix, you're a rock of the nvim community ❤️
